Source/Scene/modelMaterialsCommon.js

```diff
@@ -349,6 +349,16 @@ define([
             vertexShader += 'attribute vec4 a_joint;\n';
             vertexShader += 'attribute vec4 a_weight;\n';
         }
+        
+        if (options.addBatchIdToGeneratedShaders)
+        {
+            techniqueAttributes.a_batchId = 'batchId';
+            techniqueParameters.batchId = {
+                semantic: 'BATCHID',
+                type: WebGLConstants.UNSIGNED_SHORT
```

Comment:
> Could anybody comment whether this needs to be adjusted based on the actual data type of the batch id values in the glTF buffer?

Yes this will need to be adjusted based on the actual type in the buffer. The spec may need to be clearer in saying that  the batch id can be any type.